### PR TITLE
Download platform-agnostic Flutter Web SDK in the flutter_tool

### PR DESCRIPTION
**Pull request by harryterkelsen**
This downloads the platform-agnostic Flutter Web SDK with the locally built CanvasKit artifacts.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by christopherfujino — COMMENTED** [review]
Does this mean we are using our own version of canvaskit now?

**Comment by kevmoo** [comment]
> Does this mean we are using our own version of canvaskit now?

YES!

**Comment by christopherfujino** [comment]
> > Does this mean we are using our own version of canvaskit now?
> 
> YES!

🎆

**Comment by kevmoo** [comment]
Well, I thought so. Chats with @hterkelsen lead me to think this purpose of this PR is something different.

File: packages/flutter_tools/lib/src/flutter_cache.dart [hunk 1]
```diff
@@ -158,16 +158,13 @@ class MaterialFonts extends CachedArtifact {
 ///
 /// This SDK references code within the regular Dart sdk to reduce download size.
 class FlutterWebSdk extends CachedArtifact {
-  FlutterWebSdk(Cache cache, {required Platform platform})
-   : _platform = platform,
-     super(
+  FlutterWebSdk(Cache cache, {@Deprecated('The platform argument is deprecated') Platform? platform})
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/flutter_cache.dart:161]: Can we not just remove this field?

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/flutter_cache.dart]: For more context, as flutter_tools is not a public API, we generally do not deprecate function parameters, but just remove them (if Google testing is failing, the process would be creating a g3fix for it).

Comment (hterkelsen) [reply packages/flutter_tools/lib/src/flutter_cache.dart]: Thanks, I went back to just removing platform and will create the g3fix once the CL is available in the frob ui

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/flutter_cache.dart]: Thanks!

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
@christopherfujino Do we need to do anything about users behind firewalls? The new URL isn't that different from what we had already, but maybe the removal of chrome-infra-packages.appspot.com can break something?

**Comment by christopherfujino** [comment]
> @christopherfujino Do we need to do anything about users behind firewalls? The new URL isn't that different from what we had already, but maybe the removal of chrome-infra-packages.appspot.com can break something?

Good call-out--I think this will actually improve things, but let me test this locally

**Review by christopherfujino — APPROVED** [review]
Works for me locally, even with an artifact cache mirror configured, LGTM!

**Comment by zanderso** [comment]
SkiaPerf is reporting large reduction in compressed size on this change:

https://flutter-flutter-perf.skia.org/e/?begin=1674770761&end=1674773914&keys=Xc49c71ce49c5d3a0f673eb82db39cd79&num_commits=50&request_type=1&xbaroffset=33048

Did this PR break the size benchmark?

**Comment by christopherfujino** [comment]
> SkiaPerf is reporting large reduction in compressed size on this change:
> 
> https://flutter-flutter-perf.skia.org/e/?begin=1674770761&end=1674773914&keys=Xc49c71ce49c5d3a0f673eb82db39cd79&num_commits=50&request_type=1&xbaroffset=33048
> 
> Did this PR break the size benchmark?

per offline discussion, it looks like this is the legitimate diff between canvaskit versions.

**Pull request merged**